compiler/bindings/python/iree/compiler/tools/import_onnx/__main__.py

```diff
@@ -58,13 +60,88 @@ def main(args):
         print(m.get_asm(assume_verified=not args.no_verify))
 
 
-def load_onnx_model(file_path: Path) -> onnx.ModelProto:
-    raw_model = onnx.load(file_path)
-    inferred_model = onnx.shape_inference.infer_shapes(raw_model)
+def load_onnx_model(args: argparse.Namespace) -> onnx.ModelProto:
+    # Do shape inference two ways.  First, attempt in-memory to avoid redundant
+    # loading and the need for writing a temporary file somewhere.  If that
+    # fails, typically because of the 2 GB protobuf size limit, try again via
+    # files.  See
+    # https://github.com/onnx/onnx/blob/main/docs/PythonAPIOverview.md#shape-inference-a-large-onnx-model-2gb
+    # for details about the file-based technique.
+
+    # Make a temp dir for all the temp files we'll be generating as a side
+    # effect of infering shapes.  For now, the only file is a new .onnx holding
+    # the revised model with shapes.
+    #
+    # TODO: If the program temp_dir is None, we should be using an ephemeral
+    # temp directory instead of a hard-coded path in order to avoid data races
+    # by default.
+    input_dir = os.path.dirname(os.path.abspath(args.input_file))
+    temp_dir = (
+        Path(input_dir if args.temp_dir is None else args.temp_dir)
+        / "onnx-importer-temp"
+    )
+    shutil.rmtree(temp_dir, ignore_errors=True)
+    temp_dir.mkdir(exist_ok=True)
```

Comment:
This is sketchy code to be copying as-is. If the `--temp-dir` path is not specified, this looks like it will `rm -rf` the directory that the input file is in??

I don't see why https://docs.python.org/3/library/tempfile.html wouldn't work here? No need to have the user create their own temp dir and pass it in via a flag, and then no risk of deleting an existing directory.

Since this is a user tool for IREE, I also don't see too much value in adding options around configurable temp paths that would more facilitate torch-mlir debugging.

Reply:
If the user does not specify --temp-dir, I think it is creating an `onnx-importer-temp` directory in the directory that the input file is in

Reply:
Ah. I'd prefer to use tempfile. That's what the TODO there is suggesting anyways. It should be less code :P

Reply:
I don't think tempfile works here because  our one temporary file is being generated here: `onnx.shape_inference.infer_shapes_path(args.input_file, <path_to_output_file>, data_prop=args.data_prop)`. So, we have to pass a path where it will be stored and can't change the file being generated to a temp. But, if we don't want to deal with `temp_dir`, I can just remove that arg and just make it create a file with a certain name in the input dir and delete it at the end if the arg to remove temp files is true. What do you think?

Reply:
won't this work?
```python
# create a temporary directory using the context manager
with tempfile.TemporaryDirectory() as tmpdirname:
    print('created temporary directory', tmpdirname)
```

Reply:
Yeah temp directory would work. But, at the moment, it's only one file. Do we want to create a directory for it, or just delete the file created by the onnx call at the end.

Reply:
Keep it simple - use standard library functions with predictable behavior (create a file or directory with tempfile, use that for scratch space). I don't want to risk a bug like https://www.theregister.com/2015/01/17/scary_code_of_the_week_steam_cleans_linux_pcs/ :P

Reply:
Got it, went with the `tempfile.TemporaryDirectory`